fastlane_core/lib/fastlane_core/print_table.rb

```diff
@@ -34,16 +41,60 @@ def print_values(config: nil, title: nil, hide_keys: [], mask_keys: [])
         return params
       end
 
-      def limit_row_size(rows, max_length = 100)
+      def colorize_array(array, colors)
+        value = ""
+        array.each do  |l|
+          colored_line = l
+          colored_line = "#{colors.first[0]}#{l}#{colors.last[0]}" if colors.length > 0
+          value << colored_line
+          value << "\n"
+        end
+        return value
+      end
+
+      def transform_output(rows, transform)
         require 'fastlane_core/string_filters'
+        require 'tty-screen'
+
+        return_array = []
+        tcols = TTY::Screen.width
 
-        max_key_length = rows.map { |e| e[0].length }.max || 0
-        max_allowed_value_length = max_length - max_key_length - 7
-        rows.map do |e|
-          value = e[1]
-          value = value.to_s.truncate(max_allowed_value_length) unless [true, false].include?(value)
-          [e[0], value]
+        col_count = rows.map(&:length).first || 1
+
+        # -4 per column - as tt adds "| " and " |"
+        terminal_table_padding = 4
+        max_length = tcols - (col_count * terminal_table_padding)
+
+        max_value_length = (max_length / col_count)
+
+        rows.map do |row|
+          new_row = []
+          row.each do |col|
```

Comment:
might also be easier to read if we break the contents of this block into a `transform_column(column, truncation_style)` method?